ui/ts/models/log.ts

```diff
@@ -77,13 +76,22 @@ module Models {
         return url;
       }
 
-      constructor() {
+      constructor(nodeId?: string) {
         this.level(Utils.Format.Severity(2));
         this.max(null);
         this.startTime(null);
         this.endTime(null);
         this.pattern(null);
-        this.node(null);
+        this.node(nodeId);
+        // Add the QueryCache after initialization in order to not fetch without
+        // the node id.
+        this._data = new Utils.QueryCache((): Promise<Proto.LogEntry[]> => {
+          return m.request({ url: this._url(), method: "GET", extract: nonJsonErrors })
+            .then((results: LogResponseSet) => {
+              return results.d;
+            });
+        });
+        this.allEntries = this._data.result;
```

Comment:
with you so far. what does `this.pattern(null);` above do? is it a setter method?